tc/sgx/trusted_worker_manager/enclave_wrapper/file_io_processor.cpp

```diff
@@ -0,0 +1,298 @@
+            return FAILED;
+    }
+
+    resultStr = "FILE CLOSE SUCCESS";
+    resultSize = resultStr.length();
+    result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+    return SUCCESS;
+}
+
+// Reads file, store content of file in buffer and updates status
+// SUCCESS or FAILED in result parameter
+uint32_t FileRead(std::string fileName, uint8_t *result, size_t resultSize,
+    uint8_t *outBuf, size_t outBufSize) {
+    string resultStr;
+    if (fileName.empty()) {
+        tcf::Log(TCF_LOG_ERROR, "Filename is empty\n");
+        resultStr = "FAILED TO READ: Filename is empty";
+        resultSize = resultStr.length();
+        result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+        return FAILED;
+    }
+
+    try {
+        ifstream fileStream (fileName);
+        if (!fileStream.is_open())
+        {
+            tcf::Log(TCF_LOG_ERROR, "Unable to open file %s\n", fileName.c_str());
+            resultStr = "FAILED TO READ: Unable to open file";
+            resultSize = resultStr.length();
+            result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+            fileStream.close();
+            return FAILED;
+        }
+
+        fileStream.seekg (0, fileStream.end);
+        int fileLen = fileStream.tellg();
+        fileStream.seekg (0, fileStream.beg);
+
+        char *buff = new char[fileLen];
+        fileStream.read(buff, fileLen);
+
+        if ( !fileStream) {
+            tcf::Log(TCF_LOG_ERROR, "Failed to read complete file %s\n", fileName.c_str());
+            resultStr = "FAILED TO READ: Couldn't read complete file";
+            resultSize = resultStr.length();
+            result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+            fileStream.close();
+            return FAILED;
+        }
+
+       tcf::Log(TCF_LOG_ERROR, "File %s read has been completed\n", fileName.c_str());
+
+       // Copy the read file content to output buffer
+       outBuf = (uint8_t *)strncpy((char *)outBuf, buff, fileLen);
+       outBufSize = fileLen;
+       delete[] buff;
+       fileStream.close();
+
+    } catch (std::ifstream::failure e) {
+         tcf::Log(TCF_LOG_ERROR, "Caught exception while Reading file: %s\n", e.what());
+         resultStr = "FAILED TO READ: Caught Exception while reading file";
+         resultSize = resultStr.length();
+         result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+         return FAILED;
+    }
+
+    resultStr = "FILE READ SUCCESS";
+    resultSize = resultStr.length();
+    result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+
+    return SUCCESS;
+}
+
+// Writes file with the content of the buffer and updates status
+// SUCCESS or FAILED in result parameter
+uint32_t FileWrite(std::string fileName, uint8_t *result, size_t resultSize,
+    const uint8_t *inBuf, size_t inBufSize) {
```

Comment:
Please document `inBuf` and `inBufSize` in the function comments, as this is an external API.

Reply:
Done